### PR TITLE
julia_110: init at 1.10.0

### LINKED ISSUE #280123: Update request: julia 1.9.4 → 1.10.0
- Package name: julia
- Latest released version: 1.10.0

- Current version on the unstable channel: 1.9.4
- Current version on the stable/release channel: 1.9.4

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by NickCao**
## Description of changes
Fixes https://github.com/NixOS/nixpkgs/issues/280123

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/julia/1.10-bin.nix [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    "REPL"
+  ] ++ lib.optionals stdenv.isDarwin [
+    # Test flaky on ofborg
+    "FileWatching"
+    # Test requires pbcopy
+    "InteractiveUtils"
+    # Test requires network access
+    "Sockets"
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
+    # Test Failed at $out/share/julia/stdlib/v1.8/LinearAlgebra/test/blas.jl:702
```
Comment (thomasjm) [inline pkgs/development/compilers/julia/1.10-bin.nix:23]: This comment looks out of date as it references `v1.8`, does the test still fail?

**Review by thomasjm — COMMENTED** [review]
Thanks for doing this!

The one overall thing I notice is that there's getting to be a lot of duplication among all the `1.x.nix` and `1.x-bin.nix` files. For example, this PR marks the 6th time that the `meta` and maintainers list is duplicated. Maybe it's time soon to break common stuff into a `common.nix` file?

File: pkgs/development/compilers/julia/1.10-bin.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/development/compilers/julia/1.10-bin.nix:23]: I don't have a Darwin machine to test, and it times out on ofborg.

**Comment by NickCao** [comment]
> Maybe it's time soon to break common stuff into a common.nix file?

Great, also a good time to drop the unmaintained versions.

**Comment by NickCao** [comment]
Technically 1.9 is also EOL, but we can keep it for a little longer in case there are issues with 1.10.

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 280153` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>cantor</li>
    <li>julia</li>
    <li>julia-bin</li>
    <li>julia_19</li>
    <li>julia_19-bin</li>
    <li>labplot</li>
  </ul>
</details>

**Comment by thomasjm** [comment]
The deduplication is looking great! I notice there's also a lot of similarity between `1.9-bin.nix` and `1.10-bin.nix`, and less so with `1.6-bin.nix`. Any chance of giving them the same treatment? It's not essential I think, but would be nice to do at some point.

**Comment by NickCao** [comment]
Done. I'd like to leave `1.6-bin.nix` as is, we would get rid of it as soon as the next LTS is announced.

**Review by 7c6f434c — APPROVED** [review]
Looks good to me

**Pull request merged**